src/Microsoft.Azure.WebJobs.Host/Bindings/BindingDataPathHelper.cs

```diff
@@ -45,10 +50,21 @@ public static string ConvertParameterValueToString(object parameterValue)
                         return ((Byte)parameterValue).ToString(CultureInfo.InvariantCulture);
                     case TypeCode.SByte:
                         return ((SByte)parameterValue).ToString(CultureInfo.InvariantCulture);
+                    case TypeCode.DateTime:                        
+                        format = format ?? "yyyy-MM-ddTHH-mm-ssK"; // default to ISO 8601
+                        var dateTime = (DateTime)parameterValue;
+                        return dateTime.ToString(format, CultureInfo.InvariantCulture);
                     case TypeCode.Object:
                         if (parameterValue is Guid)
                         {
-                            return parameterValue.ToString();
+                            if (format == null)
```

Comment:
Also, as we discussed, out of scope for this PR, but it would probably be good to have an issue tracking work to refactor this so this type is not handling all of this logic.